pyproject.toml

```diff
@@ -184,6 +184,71 @@ exclude = '''
 )
 '''
 
+[tool.ruff]
+line-length = 88
+
+select = [
+  # pyflakes
+  "F",
+  # pycodestyle
+  "E",
+  "W",
+  # flake8-bugbear
+  "B",
+]
+
+ignore = [
+  # space before : (needed for how black formats slicing)
+  # "E203",  # not yet implemented
```

Comment:
It might seem that ruff doesn't implement full of flake8, since some of those rules we ignore are not implemented, but the context is that those were not prioritized, exactly because those are not needed when using black (https://github.com/charliermarsh/ruff/issues/1527#issuecomment-1368400284)